src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java

```diff
@@ -96,8 +96,14 @@ public boolean visible(String name, Map<String, Object> parsedConfig) {
 
         @Override
         public void ensureValid(String name, Object value) {
-            Object uppercaseValue = value instanceof String ? ((String) value).toUpperCase() : value;
-            if (!validValues.contains(uppercaseValue)) {
+            boolean isValidValue;
+            if(value instanceof String) {
+                isValidValue = validValues.stream().map(validValue -> validValue.toString()).anyMatch(((String) value)::equalsIgnoreCase);
+            } else {
+                isValidValue = validValues.contains(value);
+            }
+
+            if (!isValidValue) {
```

Comment:
I'm curious about this change. What was the reason for it?

Reply:
@rahulvudutala  I commented this change out locally and all the tests still passed. So it doesn't seem like this change is solving a problem.

Reply:
Reverted the code.